tmpl/wiki/tutorial-lwt.md

```diff
@@ -1,22 +1,63 @@
-[Lwt](http://www.ocsigen.org/lwt) is a lightweight cooperative threading library for OCaml. A good way to understand Lwt and its use in MirageOS is to write some simple code. This document introduces the basic concepts and suggests programs to write. Note that Lwt has a number of syntax extensions that are widely used in MirageOS. These are introduced as you go along through the tutorial. Code for all examples is in the `mirage-skeleton/lwt/src` [repository](https://github.com/mirage/mirage-skeleton/tree/master/lwt/src).
+[Lwt](http://www.ocsigen.org/lwt) is a lightweight cooperative threading library for OCaml. A good way to understand Lwt and its use in MirageOS is to write some simple code. This document introduces the basic concepts and suggests programs to write. Code for all examples is in the `mirage-skeleton/lwt/src` [repository](https://github.com/mirage/mirage-skeleton/tree/master/lwt/src).
 
-##Tutorial
+##Basics
 
-The full Lwt manual is available [elsewhere](http://ocsigen.org/lwt/manual/), but the minimal stuff needed to get started is here. The first useful function is the `return` statement, which constructs a constant thread:
+The full Lwt manual is available [elsewhere](http://ocsigen.org/lwt/manual/), but the minimal stuff needed to get started is here.
+
+The core type in Lwt is a "thread" (also known as a "promise" in some other systems).
+An `'a Lwt.t` is a thread that should produce a value of type `'a` (for example, an `int Lwt.t` should produce a single `int`).
+Initially a thread is _sleeping_ (the result is not yet known). At some point, it changes to be either _returned_ (with a value of type `'a`) or _failed_ (with an exception). Once returned or failed, a thread never changes state again.
+
+Lwt provides a number of functions for working with threads.
+The first useful function is `return`, which constructs a trivial, already-returned thread:
 
 ```
   val return: 'a -> 'a Lwt.t
 ```
 
-It is used to construct a thread that immediately returns with the provided value. Once the value is wrapped in an Lwt thread, it cannot directly be used (as the thread may not have completed yet). This is where `bind` comes in:
+This is useful if an API requires a thread, but you already happen to know the value.
+Once the value is wrapped in its Lwt thread, it cannot directly be used (as in general a thread may not have terminated yet). This is where the `>>=` operator (pronounced "bind") comes in:
 
 ```
-  val bind: 'a Lwt.t -> ('a -> 'b Lwt.t) -> 'b Lwt.t
+  val ( >>= ): 'a Lwt.t -> ('a -> 'b Lwt.t) -> 'b Lwt.t
```

Comment:
Mainly just following http://ocaml.org/learn/tutorials/guidelines.html#Lexicalconventions

> Systematically surrounding operators with spaces simplify the treatment of infix operators which are no more a complex particular case; in effect, whereas you can write `(+)` without spaces, you evidently cannot write `(*)` since `(*` is read as the beginning of a comment. You must write at least one space as in `( *)`, although an extra space after `*` is definitively preferable if you want to avoid that `*)` could be read, in some contexts, as the end of a comment. All those difficulties are easily avoided if you adopt the simple rule proposed here: keep operator symbols well separated by spaces.
In fact you will quickly feel that this rule is not so difficult to follow: the space bar is the greatest and best situated key of the keyboard, it is the easiest to enter and you cannot miss it!